src/router/authGuard.js

```diff
@@ -0,0 +1,25 @@
+import { useAuthStore } from '@/store/auth.js';
+import authRoutes from './auth';
+import userRoutes from './user';
+
+const getRouteNames = routes => {
+  const routeNames = routes.reduce((routeNames, route) => {
+    if (route.name) routeNames.push(route.name);
+    return routeNames;
+  }, []);
+  return routeNames;
+};
+
+export const authGuard = (to, _, next) => {
+  const { isLoggedIn } = useAuthStore();
+  const isAuthRoute = getRouteNames(authRoutes).includes(to.name);
+  const isUserRoute = getRouteNames(userRoutes).includes(to.name);
```

Comment:
It's much better to actually use a meta field on your routes for things like these, as the names can still change (or better yet, you might add some routes in different places which sometimes do/do not require authentication).
https://router.vuejs.org/guide/advanced/meta.html#route-meta-fields

Reply:
I'll check that and include in the next PR because it seems like I might need to take a moment to tweak the logic